src/AutoRest.CSharp/Common/Input/Configuration.cs

```diff
@@ -164,6 +164,11 @@ public enum UnreferencedTypesHandlingOption
                 }
             }
 
+            if (flavor is not null && !"azure".Equals(flavor, StringComparison.InvariantCultureIgnoreCase))
+            {
+                throw new Exception($"Invalid value of setting 'flavor': {flavor}");
```

Comment:
> Can we follow this?
> 
> ' Any other value does nothing, and therefore generates unbranded code.'

I tend to adopt strict validation here. We should not allow free input for configuration properties.

Reply:
can you give me an example of non `azure` value? I thought one flavor defines a set of code generation template. So if we only have two templates right now (e.g. `null` and `azure`), then we should not accept other values.

Reply:
Arch is discussing the different values of `flavor`. The exact values are not finalized, but it could have other values besides `azure`.

Reply:
IMHO, the forward compatibility is useless here and may cause confusion. If in the future we add `openai` flavor, the old emitter will generate unbranded codes without error. Better to just throw error so that users know they need to upgrade the emitter.

Reply:
I removed the validation since we're approaching the deadline.